domain/src/test/java/org/oppia/android/domain/exploration/ExplorationProgressControllerTest.kt

```diff
@@ -2309,17 +2420,20 @@ class ExplorationProgressControllerTest {
 
     moveToNextState()
 
-    val eventLog = fakeAnalyticsEventLogger.getMostRecentEvent()
-    assertThat(fakeAnalyticsEventLogger.getEventListCount()).isEqualTo(1)
-    assertThat(eventLog).hasStartCardContextThat {
+    val eventLogList = fakeAnalyticsEventLogger.getMostRecentEvents(2)
+    assertThat(fakeAnalyticsEventLogger.getEventListCount()).isEqualTo(2)
+    assertThat(eventLogList[0]).hasStartCardContextThat {
       hasExplorationDetailsThat().containsTestExp2Details()
       hasExplorationDetailsThat().hasStateNameThat().isEqualTo("Fractions")
       hasSkillIdThat().isEqualTo("test_skill_id_0")
     }
+    assertThat(eventLogList[1]).hasProgressSavingSuccessContextThat {
+      containsTestExp2Details()
+    }
   }
 
   @Test
-  fun testHint_offered_logsHintOfferedEvent() {
+  fun testHint_offered_logsHintOfferedEvent_logsProgressSavingSuccessEvent() {
```

Comment:
For this and the hint/solution tests below, is checkpoint state changed? I do  not understand why the progress save success event is being logged otherwise.

Reply:
Whenever there is change in hint state, for example a hint is unlocked or accessed, a corresponding checkpoint is also saved. So, a progress saving success event is logged.